docker/docker_client.go

```diff
@@ -292,7 +293,7 @@ func SearchRegistry(ctx context.Context, sys *types.SystemContext, registry, ima
 		registry = dockerV1Hostname
 	}
 
-	client, err := newDockerClientWithDetails(sys, registry, username, password, "", nil, "")
+	client, err := newDockerClientWithDetails(sys, registry, username, password, "*", nil, registry)
```

Comment:
@mtrmac what is the best way of generating the `remoteName` from the registry given by the user?

Reply:
```console
$ curl -i  https://registry-1.docker.io/v2/_catalog
HTTP/1.1 401 Unauthorized
…
Www-Authenticate: Bearer realm="https://auth.docker.io/token",service="registry.docker.io",scope="registry:catalog:*"
…
```

Also looking at https://github.com/docker/distribution/blob/4a75b72fd3621c1892143614474ff2b05d30e4b3/registry/handlers/app.go#L942-L953 , `registry:catalog:*` seems to be the scope we want.

The current `remoteName`/`actions` format can not produce a `scope="registry:…`, so the `newDockerClientWithDetails` API will need to be changed.

Maybe, to reduce the number of unnamed parameters, make `authScope` a struct of `type, name, actions string` (to follow https://github.com/docker/distribution/blob/master/docs/spec/auth/scope.md), and have a pass a single `authScope` parameter to `newDockerClientWithDetails`.  (Or there could be a single `scope string` parameter, but the callers would have to worry about formatting; well-typed data concentrates that into a single place.)

Reply:
> make `authScope` a struct of `type, name, actions string`

Or maybe even `type, name string; actions []string`, to be extra correct.